proto/viam/app/data/v1/data.proto

```diff
@@ -234,3 +252,43 @@ message RemoveAnnotationsFromBinaryDataByFilterRequest {
 message RemoveAnnotationsFromBinaryDataByFilterResponse {
   uint64 deleted_count = 1;
 }
+
+// AddTagsToBinaryDataByFileIDsRequest requests adding all specified tags to each of the files when file IDs are provided
+message AddTagsToBinaryDataByFileIDsRequest {
+  repeated string file_ids = 1;
+  repeated Annotation tags = 2;
+}
+
+// AddTagsToBinaryDataByFileIDsResponse returns nothing
+message AddTagsToBinaryDataByFileIDsResponse {}
+
+// AddTagsToBinaryDataByFilterRequest requests adding all specified tags to each of the files when a filter is provided
+message AddTagsToBinaryDataByFilterRequest {
+  Filter filter = 1;
+  repeated Annotation tags = 2;
+}
+
+// AddTagsToBinaryDataByFilterResponse returns nothing
+message AddTagsToBinaryDataByFilterResponse {}
+
+// RemoveTagsFromBinaryDataByFileIDsRequest requests removing the given tags value from each file when file IDs are provided
+message RemoveTagsFromBinaryDataByFileIDsRequest {
+  repeated string file_ids = 1;
+  repeated string value = 2;
+}
+
+// RemoveTagsFromBinaryDataByFileIDsResponse returns the number of binary files which had tags removed
+message RemoveTagsFromBinaryDataByFileIDsResponse {
+  uint64 deleted_count = 1;
```

Comment:
It feels like we either don't need this count either or we should add a count to AddTags...Request since that could also be adding multiple tags.

Reply:
AddTags will do an all-or-none insert and return an `err` if unsuccessful, so the user will know how many were added based on how many tags they had just passed in. Thus think Add should return an empty message.

`deleted_count` is more-so useful on a large scale (ByFilter) than per image (ByID) since gives back some aggregate information. This is similar to Mongo's [deleteMany](https://www.mongodb.com/docs/manual/reference/method/db.collection.deleteMany/) returning `deletedCount` and [insertMany](https://www.mongodb.com/docs/manual/reference/method/db.collection.insertMany/) returning an array of IDs (which we were going to do if we had associated IDs, but here we just have the tags).

Reply:
Sounds good!